core/src/test/java/org/apache/iceberg/TestMetadataUpdateParser.java

```diff
@@ -350,9 +485,51 @@ private static void assertEqualsAddSortOrder(
           UnboundSortOrder.UnboundSortField actualField = actual.sortOrder().fields().get(i);
           Assert.assertTrue("Fields of the sort order should be the same",
               expectedField.sourceId() == actualField.sourceId() &&
-              expectedField.nullOrder().equals(actualField.nullOrder()) &&
-              expectedField.direction().equals(actualField.direction()) &&
-              Objects.equals(expectedField.transformAsString(), actualField.transformAsString()));
+                  expectedField.nullOrder().equals(actualField.nullOrder()) &&
+                  expectedField.direction().equals(actualField.direction()) &&
+                  Objects.equals(expectedField.transformAsString(), actualField.transformAsString()));
         });
   }
+
+  private static void assertEqualsSetDefaultSortOrder(
+      MetadataUpdate.SetDefaultSortOrder expected, MetadataUpdate.SetDefaultSortOrder actual) {
+    Assert.assertEquals("Sort order id should be the same", expected.sortOrderId(), actual.sortOrderId());
+  }
+
+  // TODO - Come back to this when FileIO is injected. If this class extends TableTestBase, the validateSnapshot
+  //  methods can be reused. Also, this needs to handle V1 vs V2 for assertions. Or casting to BaseSnapshot,
+  //  BaseSnapshot::equals can be used.
+  private static void assertEqualsAddSnapshot(
+      MetadataUpdate.AddSnapshot expected, MetadataUpdate.AddSnapshot actual) {
+    Assert.fail("MetadataUpdate equality checking for AddSnapshot is not implemented yet");
+  }
```

Comment:
TODO / Follow up - Come back to this when FileIO is injected. If this class extends TableTestBase, the validateSnapshot methods can be reused. Also, this possibly needs to handle the difference between V1 vs V2 for assertions. Or casting to BaseSnapshot, BaseSnapshot::equals can be used.

Happy to remove this in this PR and handle `EqualsAddSnapshot` in another PR.

Reply:
I think this should check that the snapshot ID, manifest list location, and summary map are equal.

Reply:
I think that's sufficient.

What about the snapshots schema id, operation, and the timestamp? Possibly overkill as the three you mentioned (especially the combination of snapshot ID and manifest list location) could arguably be considered sufficient.

Reply:
I added timestamp, schema id and parent id as well as the ones you stated. That should be more than sufficient to verify that the existing parser is working (which we reuse).